### DISCUSSION TITLE
Allow the possibility to block/allow/whitelist model parameters

### DISCUSSION THREAD

**Discussion by aleixdorca** [open-webui/open-webui#6184] [General]
Hi everyone!

I was thinking it might be helpful to add a feature that allows Admins to control which model parameters users can adjust. 

Currently, users have the freedom to change any parameter, which can sometimes lead to performance issues on the local server.

For instance, if a model is set up with a specific context size to manage VRAM usage efficiently, a user may change this setting. The result might be that the model starts using the CPU instead of the GPU, potentially causing disruptions in the server's quality of service (QoS).

What do you think?

Best regards.

**Comment by tjbck** [comment]
PR welcome!

**Reply by gaby** [reply]
@tjbck This is the same issue I reported a few days ago. Here https://github.com/open-webui/open-webui/discussions/6081

We have the same issue happening almost daily. There's no way to known which user is changing model params, eventually someone changes something and Ollama will drop model from VRAM and use CPU. The system halts and we have to reboot the whole server. Basically any user can DOS the system running Open-WebUI
